packages/arb-avm-cpp/avm/include/avm/machinestate/machinestate.hpp

```diff
@@ -25,26 +25,28 @@
 #include <avm_values/value.hpp>
 #include <avm_values/vmValueParser.hpp>
 
+#include <deque>
 #include <memory>
 #include <vector>
 
+class MachineExecutionConfig;
+
 struct AssertionContext {
     std::vector<InboxMessage> inbox_messages;
     std::optional<uint256_t> next_block_height;
     size_t inbox_messages_consumed{0};
     uint256_t numSteps{0};
     uint256_t numGas{0};
-    bool blockingSideload{false};
     std::optional<value> fake_inbox_peek_value;
     std::vector<std::vector<uint8_t>> sends;
     std::vector<value> logs;
     std::vector<value> debug_prints;
+    std::deque<InboxMessage> sideloads;
```

Comment:
Just confirming, is there a reason we need a `std::duque` instead of an `std::vector`?

Reply:
You can efficiently pop from the front of it, and if you have a list of sideloads you want it's a bit of a pain to specify them in reverse order. An alternative would be keeping an index within the vector, as we do for inbox messages, but this is simpler and it doesn't need to be perfectly efficient.

Reply:
Sounds good to me